### PR TITLE
[move] Fix toml issues with move-execution

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

- The move execution crates were not included in the `external-crates/move/Cargo.toml`, leading to all sorts of issues 

## Test plan 

- CI

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

**Comment by tnowacki** [comment]
@amnn or @dariorussi, do you know if I need to update any scripts for execution cuts to reflect these changes in the future? (though we will need to fix the issues with the stdlib before changing any such scripts)

File: Cargo.toml
Comment (amnn) [inline Cargo.toml:1]: I'm assuming/hoping this was done because you flattened `v0` as well to make it easier to make other fixes (rather than it having always been broken), because this should be covered by tests...

File: external-crates/move/Cargo.toml [hunk 1]
```diff
@@ -155,6 +158,23 @@ move-vm-test-utils = { path = "crates/move-vm-test-utils" }
 move-vm-types = { path = "crates/move-vm-types" }
 prover_bytecode = { path = "crates/move-stackless-bytecode", package="move-stackless-bytecode" }
 
+# v0
```
Comment (amnn) [inline external-crates/move/Cargo.toml:161]: You don't want to make these workspace deps. Instead, we should follow the pattern that we use for other versioned crates, which is to update the toml files of the packages that depend on them to refer to them at their versioned path, but their unversioned name.

The reason for this is that it's much easier to automate (the script doesn't need to look at rust source files and find all references to the latest version and rename that to the versioned form).

See for example:

https://github.com/MystenLabs/sui/blob/127f2cb31699ebddbe1ea4935d264a235ca2e9ca/external-crates/move/move-execution/v1/crates/move-vm-runtime/Cargo.toml#L23

File: external-crates/move/Cargo.toml [hunk 2]
```diff
@@ -3,6 +3,9 @@ resolver = "2"
 
 members = [
     "crates/*",
+    "move-execution/v0/crates/*",
+    "move-execution/v1/crates/*",
+    "move-execution/v2/crates/*",
```
Comment (amnn) [inline external-crates/move/Cargo.toml:8]: Automating this is going to be ...tricky. The cut tool figures out which packages should be included or excluded from the root workspace here:

https://github.com/MystenLabs/sui/blob/127f2cb31699ebddbe1ea4935d264a235ca2e9ca/sui-execution/cut/src/plan.rs#L406

But it doesn't currently understand how to handle multiple workspaces, ...or glob patterns. One thing you could do is piggyback off the logic we have in the wrapper script, that updates TOML files when a new cut is introduced, here:

https://github.com/MystenLabs/sui/blob/127f2cb31699ebddbe1ea4935d264a235ca2e9ca/scripts/execution_layer.py#L420

This looks for comments in TOML files that contain "$CUT" and adds a line above them with "$CUT" replaced with the cut name, see e.g.

https://github.com/MystenLabs/sui/blob/127f2cb31699ebddbe1ea4935d264a235ca2e9ca/sui-execution/Cargo.toml#L18-L32

This is maybe a little more brittle, but I think it's the fastest way to get unblocked without losing the automation support.

File: external-crates/move/move-execution/v0/crates/bytecode-verifier-libfuzzer/Cargo.toml
Comment (amnn) [inline external-crates/move/move-execution/v0/crates/bytecode-verifier-libfuzzer/Cargo.toml:1]: Two comments here: I'm guessing that this crate is not included in the list of crates to cut right now, here:

https://github.com/MystenLabs/sui/blob/127f2cb31699ebddbe1ea4935d264a235ca2e9ca/scripts/execution_layer.py#L354-L366

But we should do that to make sure all future cuts get this crate and it's properly fixed up (Adding a new `-p` flag with the crate name should suffice).

Also as per the earlier comment about its dependencies, this is where we would need to change the `move-bytecode-verifier` dependency from a workspae dependency to a `path` dependency with a customised package name (see my earlier comment on the workspace `Cargo.toml`).

File: external-crates/move/move-execution/v0/crates/bytecode-verifier-libfuzzer/fuzz_targets/code_unit.rs
Comment (amnn) [inline external-crates/move/move-execution/v0/crates/bytecode-verifier-libfuzzer/fuzz_targets/code_unit.rs:1]: ...and then all the source files can be left alone.

**Review by amnn — COMMENTED** [review]
Amazing, thanks for tackling this. I've left comments on how to align this with how the cut script automates this process, so that future cuts all get tests and are included in the workspace.

**Comment by tnowacki** [comment]
@amnn thanks for all the guidance! 
I _think_ I have addressed the concerns, but am not entirely convinced I did it correctly.

File: external-crates/move/Cargo.toml
Comment (amnn) [inline external-crates/move/Cargo.toml:1]: In this file, the changes to `members` is correct, but I don't think you should need the changes to workspace dependencies -- generally, if a crate is versioned in that way, we other crates should depend on it by path, or through some multiplexing crate (which we don't have within the Move workspace, but for `sui` it's the `sui-execution` crate).

**Review by amnn — APPROVED** [review]
LGTM!  Note the comment about workspace deps -- they are fine to be there, but if you *needed* to add them, then that's a worrying sign (probably something is set up incorrectly, bypassing versioning), so it's probably best to remove them.

File: external-crates/move/move-execution/v0/crates/invalid-mutations/src/bounds.rs
[the remark this answers is not in the document]
Comment (tnowacki) [reply external-crates/move/move-execution/v0/crates/invalid-mutations/src/bounds.rs:1]: I tried to do git mv with then restoring the previous file. This was the result :( 

Not sure if there is anything else to do

File: external-crates/move/Cargo.toml
Comment (tnowacki) [reply external-crates/move/Cargo.toml:1]: I tried to remove them at one point, but it did not work. Let me try again now that things are all good across the board

Comment (amnn) [reply external-crates/move/Cargo.toml:1]: I can take a look tomorrow as well, thanks for these fixes!

Comment (amnn) [reply external-crates/move/Cargo.toml:1]: (...never mind, I see it's fixed! Sorry, got confused by Github)
